### PR TITLE
Feature/25/charts

### PR DESCRIPTION
**Pull request by zakrzaq**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -15,6 +17,8 @@ const App = () => {
       <AppWrapper>
         <AppRoutes />
         <ModalExample />
+        <MontlyExpenseIncome />
```
Comment (m-ossolinski) [inline src/App.js:20]: `MonthlyTransactionsChart`

File: src/App.js [hunk 2]
```diff
@@ -15,6 +17,8 @@ const App = () => {
...
+        <CategoryExpense />
```
Comment (m-ossolinski) [inline src/App.js:21]: `CategoriesPieChart`

File: src/components/Charts/CategoryExpense.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
+import { Pie } from "react-chartjs-2";
+import transactionsService from "../../services/api/transactionsService.";
+import { useState, useEffect } from "react/cjs/react.development";
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:5]: move to top

File: src/components/Charts/CategoryExpense.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+ChartJS.register(ArcElement, Tooltip, Legend);
+
+export const CategoryExpense = () => {
+  const [transactions, setTransactions] = useState([]);
+  const [categories, setCategories] = useState([]);
+  const [sums, setSums] = useState(0);
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:12]: `categorySums, setCategorySums`

File: src/components/Charts/CategoryExpense.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
+import { Pie } from "react-chartjs-2";
+import transactionsService from "../../services/api/transactionsService.";
+import { useState, useEffect } from "react/cjs/react.development";
+
+ChartJS.register(ArcElement, Tooltip, Legend);
+
+export const CategoryExpense = () => {
+  const [transactions, setTransactions] = useState([]);
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:10]: `transactionsList, setTransactionsList`

File: src/components/Charts/CategoryExpense.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  const [categories, setCategories] = useState([]);
+  const [sums, setSums] = useState(0);
+  const getTransactions = async () => {
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:13]: `getTransactionsList`

File: src/components/Charts/CategoryExpense.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    try {
+      const response = await transactionsService.getTransactionList();
+      setTransactions(response);
+      const transactionCategories = response.map((item) => item.category);
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:17]: add helper function for extracting categories types

File: src/components/Charts/CategoryExpense.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+      setCategories(
+        transactionCategories.filter((x, i, a) => a.indexOf(x) == i)
+      );
+    } catch (error) {
+      throw new Error("Transaction list could not been shown.");
+    } finally {
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:23]: prob some loader should be switched off here ;d

File: src/components/Charts/CategoryExpense.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    }
+  };
+
+  useEffect(() => {
+    getTransactions();
+
+    // TODO - FILTER OUT INCOME FORM CATEGORIES
+    let categorySums = [];
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:31]: `sumForEachCategory`

File: src/components/Charts/CategoryExpense.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    categories.forEach((cat) => {
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:32]: how about dog ;d

File: src/components/Charts/CategoryExpense.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+        (trans) => trans.category === cat
+      );
+      categorySums.push(
+        categoryArray.reduce((acc, item) => acc + parseInt(item.amount), 0)
+      );
+    });
+    setSums(categorySums);
+  }, []);
+
+  const options = {
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:43]: `chartOptions`, `chartSettings`

File: src/components/Charts/CategoryExpense.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    "rgba(255, 99, 132, 0.5)",
+    "rgba(53, 162, 235, 0.5)",
+    "rgba(53, 50, 170, 0.5)",
+    "rgba(53, 162, 100, 0.5)",
+    "rgba(250, 162, 235, 0.5)",
+    // TODO - NEED MORE COLOURS DEFINED
+  ];
+
+  const data = {
+    labels,
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:68]: can't just be `labels: categories`?

File: src/components/Charts/CategoryExpense.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+    datasets: [
+      {
+        label: "Expense Chart",
+        data: inputData,
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:72]: `data: sums`

File: src/components/Charts/MontlyExpenseIncome.js [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+      position: "top",
+    },
+    title: {
+      display: true,
+      text: "Monthly Expense vs Income",
+    },
+  },
+};
+
+const labels = [
```
Comment (m-ossolinski) [inline src/components/Charts/MontlyExpenseIncome.js:35]: should be moved to charts helpers/utils file

File: src/components/Charts/CategoryExpense.js [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+      title: {
+        display: true,
+        text: "Expense by Category",
+      },
+    },
+  };
+
+  const labels = categories;
+  const inputData = sums;
+  const pieChartColours = [
```
Comment (m-ossolinski) [inline src/components/Charts/CategoryExpense.js:58]: should be moved to charts helpers/utils file

**Review by m-ossolinski — CHANGES_REQUESTED** [review]

File: src/components/Charts/CategoryExpense.js (hunk 10 above)
Comment (zakrzaq) [reply src/components/Charts/CategoryExpense.js:32]: i prefer fish :p

File: src/components/Charts/CategoryExpense.js (hunk 8 above)
Comment (zakrzaq) [reply src/components/Charts/CategoryExpense.js:23]: Not sure what you mean?

**Review by m-ossolinski — APPROVED** [review]

**Review by hannakozak — APPROVED** [review]

**Pull request merged**